components/automate-ui/src/app/entities/users/userself.effects.ts

```diff
@@ -0,0 +1,111 @@
+import { catchError, map, mergeMap, filter } from 'rxjs/operators';
+import { Injectable } from '@angular/core';
+import { HttpErrorResponse } from '@angular/common/http';
+import { Actions, Effect, ofType } from '@ngrx/effects';
+import { of, combineLatest } from 'rxjs';
+import { NgrxStateAtom } from 'app/ngrx.reducers';
+import { Store } from '@ngrx/store';
+import { identity } from 'lodash/fp';
+
+import { userSelfId } from './userself.selectors';
+import { CreateNotification } from 'app/entities/notifications/notification.actions';
+import { Type } from 'app/entities/notifications/notification.model';
+import {
+  UserSelfActionTypes,
+  GetUserSelf,
+  GetUserSelfSuccess,
+  GetUserSelfFailure,
+  UpdatePasswordSelf,
+  UpdateNameSelf,
+  UpdatePasswordSelfSuccess,
+  UpdateNameSelfSuccess,
+  UpdateNameSelfFailure,
+  UpdatePasswordSelfFailure
+} from './userself.actions';
+import { UserRequests } from './user.requests';
+import { SelfUser } from './userself.model';
+import { iamMajorVersion } from 'app/entities/policies/policy.selectors';
+import { IAMMajorVersion } from 'app/entities/policies/policy.model';
+
+@Injectable()
+export class UserSelfEffects {
+  constructor(
+    private actions$: Actions,
+    private requests: UserRequests,
+    private store$: Store<NgrxStateAtom>
+  ) { }
+
+  @Effect()
+  getUserSelf$ = combineLatest([
+    this.actions$.pipe(ofType<GetUserSelf>(UserSelfActionTypes.GET)),
+    this.store$.select(userSelfId).pipe(filter(identity))]).pipe(
+      mergeMap(([_action, userId]) =>
+        this.requests.getUser(userId).pipe(
+        map((resp: SelfUser) => new GetUserSelfSuccess(resp)),
+        catchError((error: HttpErrorResponse) => of(new GetUserSelfFailure(error)))))
+      );
+
+  @Effect()
+  getUserSelfFailure$ = this.actions$.pipe(
+    ofType(UserSelfActionTypes.GET_FAILURE),
+    // when first logging in, we see this failure with no error.
```

Comment:
🤔 does this always happen?

Reply:
Yeah, it always happens on log in.

Reply:
I think it has something to do with the signin.component setting the user self ID which kicks off getting the user self before automate is fully started. I am guessing the request to get user self is canceled.